packages/lib/eventManager.ts

```diff
@@ -20,30 +20,34 @@ export enum EventName {
 	NoteResourceIndexed = 'noteResourceIndexed',
 }
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any -- Partial refactor of old code from before rule was applied
+export type EventListenerCallback = (...args: any[])=> void;
+// eslint-disable-next-line @typescript-eslint/no-explicit-any -- Partial refactor of old code from before rule was applied
+type AppStateChangeCallback = (event: { value: any })=> void;
+// eslint-disable-next-line @typescript-eslint/no-explicit-any -- Partial refactor of old code from before rule was applied
+type FilterObject = any;
```

Comment:
Currently, we lack types for different events. One solution could be to use [an indexed-access type](https://www.typescriptlang.org/docs/handbook/2/indexed-access-types.html) to store and look up types for the different events.


For example,
```ts
enum EventName {
	ResourceCreate = 'resourceCreate',
	ResourceChange = 'resourceChange',
	SettingsChange = 'settingsChange',
	...
}

interface ResourceCreateEvent {
	// Some data here
}

export type EventTypes = {
	[EventName.ResourceCreate]: ResourceCreateEvent,
	[EventName.ResourceChange]: ResourceChangeEvent,
	...
};

type EventListener<T> = (event: T)=>void;
```

The `on` method could then look similar to the following:
```ts
	public on<T extends EventName>(eventName: T, callback: EventListener<EventTypes[T]>) {
```